index.js

```diff
@@ -55,10 +56,26 @@ function SRIHashAssets(inputNodes, options) {
 SRIHashAssets.prototype = Object.create(CachingWriter.prototype);
 SRIHashAssets.prototype.constructor = SRIHashAssets;
 
+SRIHashAssets.prototype.getBaseHREF = function getBaseHREF(string) {
+  var baseTag = string.match(BASE_CHECK);
+  if (baseTag && baseTag[0]) {
+    var href = baseTag[0].match(HREF_CHECK);
+    var relativePath = href && href[1];
+
+    if (!relativePath) { return null; }
+
+    return this.inputPaths[0] + relativePath;
```

Comment:
OK, I added a guard to ensure that we do not use a `<base>` that doesn't start with `/` (which should prevent escaping out).